### ISSUE TITLE
Fully embrace the Conventional Commits

### ISSUE THREAD

**Issue by Ornias1993** [openzfs/zfs#10852]
### Describe the feature would like to see added to OpenZFS

It would be nice to see a full implementation of the Conventional Commits specification as described here:
https://www.conventionalcommits.org/

This includes style checking, documenting the commit requirements and documenting when certain header tags should be used (such as `feat:` )

### How will this feature improve OpenZFS?

When doing research on automatic changelog generation, I came to the conclusion that the easiest option would be to use commit titles. However: We don't just want to add all commit titles. 
Currently, as @behlendorf described in https://github.com/openzfs/zfs/issues/10756#issuecomment-677854767, this meant:

> we're going to need to go through the commit history to identify noteworthy smaller changes.

Even without automation this is unworkable and unreliable.

We need a standard that is both reliable and easy to use, to filter commits more easily for addition to the changelog. Preferable based on the commit header.
Conventional Commits seem to be that standard for a few reasons:

- The only added change to our current workflow is the added "type" to the header
- It provides a clear guideline to vet commits messages against.
- It is supposed to be as machine readable as possible

### Additional context

TLDR: applying this standard, changes nothing to the current workflow, except allowing headertags in commit messages and limiting what is allowed in a commit header. Both are not an issue at all.

Note:
This can also be used to auto-tag PR's based on commit headers and to auto-generate changelogs, but both are not part of this issue

**Comment by Ornias1993** [comment]
Parts of this have been applied by now, others are not likely to be applied or most likely will be part of a larger procedure overhaul, if it ever happens.
